pkg/ccl/backupccl/backup_job.go

```diff
@@ -610,23 +625,27 @@ func (b *backupResumer) readManifestOnResume(
 	// they could be using either the new or the old foreign key
 	// representations. We should just preserve whatever representation the
 	// table descriptors were using and leave them alone.
-	desc, memSize, err := readBackupManifest(ctx, mem, defaultStore, backupManifestCheckpointName,
+	// Read in checkpoint from the /progress folder, if it isn't there
+	// because it is an older backup, that is okay because we will
+	// throw an error and restart the backup from scratch.
+	desc, memSize, err := readBackupManifest(ctx, mem, defaultStore, progressDirectory+"/"+backupManifestCheckpointName,
 		details.EncryptionOptions)
 
 	if err != nil {
 		if !errors.Is(err, cloud.ErrFileDoesNotExist) {
 			return nil, 0, errors.Wrapf(err, "reading backup checkpoint")
 		}
+
 		// Try reading temp checkpoint.
-		tmpCheckpoint := tempCheckpointFileNameForJob(b.job.ID())
+		tmpCheckpoint := progressDirectory + "/" + tempCheckpointFileNameForJob(b.job.ID())
```

Comment:
Similarly, should the prefix `progressDirectory/` go into the `tempCheckpointFileNameForJob` function itself?

Reply:
I made some helper functions for reading/writing and now progressDirectory+"/"+... is being used way less but still a few times since it can be either progressDirectory+"/"+checkpoint or progressDirectory+"/"+temp-checkpoint. I could make a const for both if you think it's worth it for readability though